Dockerfile.arm64

```diff
@@ -11,8 +11,8 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG QEMU_IMAGE=calico/go-build:v0.55
-FROM ${QEMU_IMAGE} as qemu
+ARG QEMU_IMAGE
+FROM ${QEMU_IMAGE}-arm64 as qemu
```

Comment:
In order to tag the image with correct achitecture we are using `--platform` with the `docker build` command causing docker to pull the `arm64` image to the local repository and tag it as `v0.55` then docker uses this localy stored image to build all other architectures causing an issue in binary build step.

Reply:
I'm sorry, I don't understand.  Why do we need to add "-arm64" here, but we don't do that for the other architectures?

> In order to tag the image

Which image do you mean?

> with correct architecture we are using --platform with the docker build command causing docker to pull the arm64 image

Which image do you mean here?

> to the local repository and tag it as v0.55

Huh?  It should only be calico-build that has the v0.55 tag, but I don't think you mean the calico-build image here, do you?

> then docker uses this locally stored image to build all other architectures causing an issue in binary build step.

Reply:
> Why do we need to add "-arm64" here, but we don't do that for the other architectures?

It is because of the `--platform=linux/arm64/v8` pulls an `arm64` image and tags it as both `calico/go-build:v0.55` and `calico/go-build:v0.55-arm64`.


In `arm64` dockerfile `FROM SCRATCH` pulls an empty `amd64` layer in the build process which results in an incorrect architecture tag for the output image (aka `calico/ctl`).`--platform` prevents docker from tagging image (`calico/ctl`) architecture as `amd64`.

Here is an example from cni repo:
https://github.com/projectcalico/cni-plugin/blob/f6cd1b86daec16b04e095b04abf0af3bffb73a89/Dockerfile.arm64#L1
```
docker inspect calico/cni:v3.20.1-arm64 | egrep -i arch
        "Architecture": "amd64",
```

> Which image do you mean?

Output image aka `calico/ctl:vxxxxx-arm64`


> Which image do you mean here?

`calico/go-build:v0.55-arm64` will be pulled and stored as both `calico/go-build:v0.55` and `calico/go-build:v0.55-arm64`

> Huh? It should only be calico-build that has the v0.55 tag, but I don't think you mean the calico-build image here, do you?

No, I meant `calico/go-build:v0.55-arm64` will be saved as `calico/go-build:v0.55`
```
root@build-dummy:~/calicoctl# docker image ls
REPOSITORY   TAG       IMAGE ID   CREATED   SIZE
root@build-dummy:~/calicoctl# docker pull --platform=linux/arm64/v8 calico/go-build:v0.55 &> /dev/null
root@build-dummy:~/calicoctl# docker image ls
REPOSITORY        TAG       IMAGE ID       CREATED      SIZE
calico/go-build   v0.55     633df77ba924   5 days ago   1.74GB
root@build-dummy:~/calicoctl# docker inspect 633df77ba924 | egrep -i arch
        "Architecture": "arm64",
```